src/SFA.DAS.PAS.Account.Api/Orchestrator/EmailOrchestrator.cs

```diff
@@ -15,65 +17,111 @@ public class EmailOrchestrator
         private readonly IAccountOrchestrator _accountOrchestrator;
         private readonly IMediator _mediator;
         private readonly IIdamsEmailServiceWrapper _idamsEmailServiceWrapper;
-        private readonly ProviderApprenticeshipsServiceConfiguration _configuration;
+        private readonly IProviderCommitmentsLogger _logger;
 
-        public EmailOrchestrator(IAccountOrchestrator accountOrchestrator, IMediator mediator, IIdamsEmailServiceWrapper idamsEmailServiceWrapper, ProviderApprenticeshipsServiceConfiguration configuration)
+        public EmailOrchestrator(IAccountOrchestrator accountOrchestrator, IMediator mediator, IIdamsEmailServiceWrapper idamsEmailServiceWrapper, IProviderCommitmentsLogger logger)
         {
             _accountOrchestrator = accountOrchestrator;
             _mediator = mediator;
             _idamsEmailServiceWrapper = idamsEmailServiceWrapper;
-            _configuration = configuration;
+            _logger = logger;
         }
 
-        public async Task SendEmailToAllProviderRecipients(long ukprn, ProviderEmailRequest message)
+        public async Task SendEmailToAllProviderRecipients(long providerId, ProviderEmailRequest message)
         {
-            List<string> recipients = new List<string>();
+            var recipients = new List<string>();
 
-            if (!_configuration.CommitmentNotification.UseProviderEmail)
+            Task<List<string>> idamsUsersTask;
+            Task<List<string>> idamsSuperUsersTask;
+
+            var idamsError = false;
+            try
+            {
+                _logger.Info($"Retrieving DAS Users and Super Users from Provider IDAMS for Provider {providerId}");
+                idamsUsersTask = _idamsEmailServiceWrapper.GetEmailsAsync(providerId);
+                idamsSuperUsersTask = _idamsEmailServiceWrapper.GetSuperUserEmailsAsync(providerId);
+
+                await Task.WhenAll(idamsUsersTask, idamsSuperUsersTask);
+            }
+            catch (Exception ex)
             {
-                recipients = _configuration.CommitmentNotification.ProviderTestEmails;
+                idamsError = true;
+                _logger.Error(ex, "An error occurred retrieving users from Provider IDAMS");
+                idamsUsersTask = Task.FromResult(new List<string>());
+                idamsSuperUsersTask = Task.FromResult(new List<string>());
             }
 
+            var idamsUsers = await idamsUsersTask;
+            var idamsSuperUsers = await idamsSuperUsersTask;
+            var allIdamsUsers = idamsUsers.Concat(idamsSuperUsers).Distinct().ToList();
+            _logger.Info($"{allIdamsUsers.Count} total users retrieved from IDAMS for Provider {providerId} ({idamsUsers.Count} DAS Users; {idamsSuperUsers.Count} Super Users)");
+
```

Comment:
The code just above here I found hard to determine what it was doing. Could we extract it into a function such as 

``` 
private async Task<(bool Error, List<string> Users)> GetIdamsUsers(long providerId)
        {
            Task<List<string>> idamsUsersTask;
            Task<List<string>> idamsSuperUsersTask;

            var idamsError = false;
            try
            {
                _logger.Info($"Retrieving DAS Users and Super Users from Provider IDAMS for Provider {providerId}");
                idamsUsersTask = _idamsEmailServiceWrapper.GetEmailsAsync(providerId);
                idamsSuperUsersTask = _idamsEmailServiceWrapper.GetSuperUserEmailsAsync(providerId);

                await Task.WhenAll(idamsUsersTask, idamsSuperUsersTask);
            }
            catch (Exception ex)
            {
                idamsError = true;
                _logger.Error(ex, "An error occurred retrieving users from Provider IDAMS");
                idamsUsersTask = Task.FromResult(new List<string>());
                idamsSuperUsersTask = Task.FromResult(new List<string>());
            }

            var idamsUsers = await idamsUsersTask;
            var idamsSuperUsers = await idamsSuperUsersTask;
            var allIdamsUsers = idamsUsers.Concat(idamsSuperUsers).Distinct().ToList();
            _logger.Info($"{allIdamsUsers.Count} total users retrieved from IDAMS for Provider {providerId} ({idamsUsers.Count} DAS Users; {idamsSuperUsers.Count} Super Users)");
            return (idamsError, allIdamsUsers)l
        } 
```

It gives an explicit intent to the action being performed.

Reply:
Done!